### PR TITLE
Publisher Library can swap files

### LINKED ISSUE #5780: Replace files in Publisher Library
A hosting client has brought to our attention that there is no simple way for a Journal Manager to replace a file and keep the same URL.  

Deleting and re-adding the file causes the numbering at the end to change. The only way this can be done is by replacing the file in the backend in order to keep the URL the same. 

It would be ideal to keep the same URL if this is hyperlinked in different pages of the journal.

### PR DESCRIPTION
**Pull request by wopsononock**
Fixes #5780

### CODE REVIEW

**Comment by wopsononock** [comment]
This PR supersedes #9482, which had problems with my commit name and email that I couldn't resolve without starting over. This addresses the code issues raised there and tests successfully with stable-3_4_0.

File: classes/file/PKPLibraryFileManager.php [hunk 1]
```diff
@@ -112,16 +112,47 @@ public function &copyFromTemporaryFile(&$temporaryFile, $libraryFileType)
...
+
+        return $libraryFile;
+    }
+
+    /**
+     * Routine to replace a library file from a temporary file.
+     * @param $libraryFileType int LIBRARY_FILE_TYPE_...
+     * @return LibraryFile|false the updated LibraryFile, or false on error
+     */
+    function replaceFromTemporaryFile(object &$temporaryFile, int $libraryFileType, LibraryFile $libraryFile) {
```
Comment (asmecher) [inline classes/file/PKPLibraryFileManager.php:128]: `object` should be `TemporaryFile` (and a `use` statement for that should be added at the top).

Comment (asmecher) [reply classes/file/PKPLibraryFileManager.php:128]: Also, the reference isn't necessary here.

File: classes/file/PKPLibraryFileManager.php [hunk 2]
```diff
@@ -112,16 +112,47 @@ public function &copyFromTemporaryFile(&$temporaryFile, $libraryFileType)
...
+
+        if ($originalServerFilename !== $libraryFile->getServerFileName()) {
+                unlink($this->getBasePath() . $originalServerFilename);
+        }
+        return $libraryFile;
+    }
+
+    /**
+     * Routine to assign metadata to a library file from a temporary file
+     * @param $temporaryFile object
```
Comment (asmecher) [inline classes/file/PKPLibraryFileManager.php:144]: Should be `TemporaryFile`, not `object`

File: classes/file/PKPLibraryFileManager.php [hunk 3]
```diff
@@ -112,16 +112,47 @@ public function &copyFromTemporaryFile(&$temporaryFile, $libraryFileType)
...
+     * @param $libraryFileType int LIBRARY_FILE_TYPE_...
+     * @param $libraryFile LibraryFile 
+     * @return LibraryFile the updated LibraryFile
+     */
+    function &assignFromTemporaryFile(&$temporaryFile, $libraryFileType, $libraryFile) {
```
Comment (asmecher) [inline classes/file/PKPLibraryFileManager.php:149]: References are unnecessary here.

**Review by asmecher — CHANGES_REQUESTED** [review]
Just a few small changes to request, thanks!

**Comment by wopsononock** [comment]
@asmecher I made the changes you requested.

**Pull request merged**